src/trio/_util.py

```diff
@@ -203,6 +203,23 @@ def _return_value_looks_like_wrong_library(value: object) -> bool:
     return coro
 
 
+class EmptySuperclass:
+    """Used as a runtime placeholder for classes which should be imported only during TYPE_CHECKING.
+
+    This can be subscripted with anything, then gets discarded entirely from the base classes list.
+    """
+
+    def __getitem__(self, item: object) -> EmptySuperclass:
```

Comment:
(nitpick) really this is `Self`.

Reply:
Probably should have made this final, but actually I don't need it since I can remove `Generic`.